presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFunction.java

```diff
@@ -23,33 +22,28 @@
 import com.facebook.presto.spi.function.TypeParameter;
 import com.facebook.presto.spi.type.Type;
 import com.facebook.presto.type.TypeUtils;
-import com.google.common.collect.ImmutableList;
 import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
 import it.unimi.dsi.fastutil.longs.LongSet;
 
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
 
-@ScalarFunction("array_distinct")
+@ScalarFunction(value = "array_distinct", deterministic = true)
```

Comment:
The default for `ScalarFunction.deterministic()` is true so this is not necessary.